parquet-variant/src/builder.rs

```diff
@@ -592,12 +592,16 @@ impl<'a> ObjectBuilder<'a> {
     }
 
     /// Add a field with key and value to the object
-    pub fn append_value<'m, 'd, T: Into<Variant<'m, 'd>>>(&mut self, key: &str, value: T) {
+    ///
+    /// Note: when inserting duplicate keys, the new value overwrites the previous mapping,
+    /// but the old value remains in the buffer, resulting in a larger variant
+    pub fn insert<'m, 'd, T: Into<Variant<'m, 'd>>>(&mut self, key: &str, value: T) {
         let field_id = self.metadata_builder.add_field_name(key);
         let field_start = self.buffer.offset();
+
+        self.fields.insert(field_id, field_start);
+
         self.buffer.append_value(value);
```

Comment:
```suggestion
        self.fields.insert(field_id, field_start);
        self.buffer.append_value(value);
```
(the extra whitespace didn't seem very helpful)